modules/datastore/src/PostImportResultFactory.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Drupal\datastore;
+
+use Drupal\Core\Database\Connection;
+use Drupal\metastore\ResourceMapper;
+use Drupal\common\DataResource;
+
+/**
+ * Factory class to create PostImportResult objects.
+ */
+class PostImportResultFactory {
+
+  /**
+   * The database connection.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected Connection $connection;
+
+  /**
+   * The metastore resource mapper service.
+   *
+   * @var \Drupal\metastore\ResourceMapper
+   */
+  protected ResourceMapper $resourceMapper;
+
+  /**
+   * Constructs a PostImportResultFactory instance.
+   *
+   * @param \Drupal\Core\Database\Connection $connection
+   *   The database connection.
+   * @param \Drupal\metastore\ResourceMapper $resourceMapper
+   *   The resource mapper service.
+   */
+  public function __construct(Connection $connection, ResourceMapper $resourceMapper) {
+    $this->connection = $connection;
+    $this->resourceMapper = $resourceMapper;
+  }
+
+  /**
+   * Creates a PostImportResult instance.
+   *
+   * Passes status, message and the resource when initialized.
+   *
+   * @param string $status
+   *   Status of the post import process.
+   * @param string $message
+   *   Messages retrieved during the post import process.
+   * @param \Drupal\common\DataResource $resource
+   *   The DKAN resource being imported.
+   *
+   * @return \Drupal\datastore\PostImportResult
+   *   The PostImportResult object.
+   */
+  public function initializeFromResource($status, $message, DataResource $resource): PostImportResult {
+    return new PostImportResult(
+      $status,
+      $message,
+      $resource,
+      $this->connection,
+    );
+  }
+
+  /**
+   * Creates a PostImportResult instance.
+   *
+   * Passes the dataset info when initialized.
+   *
+   * @param array $datasetInfo
+   *   The dataset info.
+   *
+   * @return \Drupal\datastore\PostImportResult
+   *   The PostImportResult object.
+   */
+  public function initializeFromDatasetInfo(array $datasetInfo): PostImportResult {
```

Comment:
I think this and the docbloc are a little misleading because you don't actually take the full DatasetInfo array, you're looking for a specific Distribution array, right?